src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,376 @@
+          const price = bnOrZero(tvl)
+            .div(bnOrZero(totalSupply.toString()).div(`1e${lpAssetPrecision}`))
+            .toString()
+          const lpMarketData = {
+            [foxEthLpAssetId]: { price, marketCap: '0', volume: '0', changePercent24Hr: 0 },
+          }
+          // hacks for adding lp price and price history
+          dispatch(marketData.actions.setCryptoMarketData(lpMarketData))
+          Object.values(HistoryTimeframe).forEach(timeframe => {
+            dispatch(
+              marketData.actions.setCryptoPriceHistory({
+                data: [{ price: bnOrZero(price).toNumber(), date: 0 }],
+                args: { timeframe, assetId: foxEthLpAssetId },
+              }),
+            )
+          })
+          const data = { accountAddress, tvl, apy, isLoaded: true }
+          dispatch(foxEth.actions.upsertLpOpportunity(data))
+          return { data }
+        } catch (err) {
+          moduleLogger.error(err, 'getFoxEthLpMetrics')
+          return {
+            error: {
+              error: 'LP metrics is not loaded',
+              status: 'CUSTOM_ERROR',
+            },
+          }
+        }
+      },
+    }),
+    getFoxEthLpAccountData: build.query<GetFoxEthLpAccountDataReturn, GetFoxEthLpAccountDataArgs>({
+      queryFn: async ({ accountAddress }, injectedStore) => {
+        try {
+          const { getState, dispatch } = injectedStore
+          const state: any = getState() // ReduxState causes circular dependency\
+          const assets: AssetsState = state.assets
+          const marketDataState: MarketDataState = state.marketData
+          const ethPrecision = assets.byId[ethAssetId].precision
+          const foxPrecision = assets.byId[foxAssetId].precision
+          const lpAssetPrecision = assets.byId[foxEthLpAssetId].precision
+          const lpTokenPrice = marketDataState.crypto.byId[foxEthLpAssetId]?.price ?? '0'
+          const uniV2LPContract = getOrCreateContract(
+            UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+            IUniswapV2Pair.abi,
+          )
+          const balance = await uniV2LPContract.balanceOf(accountAddress)
+          const cryptoAmount = bnOrZero(balance.toString()).div(bn(10).pow(lpAssetPrecision))
+          const fiatAmount = cryptoAmount.times(lpTokenPrice).toFixed(2)
+          const totalSupply = await uniV2LPContract.totalSupply()
+          const reserves = await uniV2LPContract.getReserves()
+
+          const userOwnershipOfPool = bnOrZero(balance.toString()).div(
+            bnOrZero(totalSupply.toString()),
+          )
+          const underlyingEthAmount = userOwnershipOfPool
+            .times(bnOrZero(reserves[0].toString()))
+            .div(`1e${ethPrecision}`)
+            .toString()
+          const underlyingFoxAmount = userOwnershipOfPool
+            .times(bnOrZero(reserves[1].toString()))
+            .div(`1e${foxPrecision}`)
+            .toString()
+
+          const data = {
```

Comment:
we should be upserting this stuff into the portfolio too, treating it like any other asset with a balance